storm-client/src/jvm/org/apache/storm/messaging/TransportFactory.java

```diff
@@ -37,7 +38,7 @@ public static IContext makeContext(Map<String, Object> topoConf) {
                 //case 1: plugin is a IContext class
                 transport = (IContext) obj;
                 //initialize with storm configuration
-                transport.prepare(topoConf);
+                transport.prepare(topoConf, metricRegistry);
             } else {
                 //case 2: Non-IContext plugin must have a makeContext(topoConf) method that returns IContext object
                 Method method = klass.getMethod("makeContext", Map.class);
```

Comment:
I am fine with simply ignoring the metricRegistry when the IContext is created this way, and it can be added by invoking a different `makeContext` method that includes `StormMetricRegistry` as a parameter in the future if desired. 

Can we add some comments like "StormMetricRegistry is ignored if IContext is created this way" here so it might help future developers? Thanks